d2d1: emit no join geometry for collinear outline segments.

diff --git a/dlls/d2d1/geometry.c b/dlls/d2d1/geometry.c
--- a/dlls/d2d1/geometry.c
+++ b/dlls/d2d1/geometry.c
@@ -2929,6 +2929,10 @@ static BOOL d2d_geometry_outline_add_join(struct d2d_geometry *geometry,
     size_t base_idx;
     float ccw;
 
+    ccw = d2d_point_ccw(&origin, prev, next);
+    if (ccw == 0.0f)
+        return TRUE;
+
     if (!d2d_array_reserve((void **)&geometry->outline.vertices, &geometry->outline.vertices_size,
             geometry->outline.vertex_count + 4, sizeof(*geometry->outline.vertices)))
     {
@@ -2946,17 +2950,7 @@ static BOOL d2d_geometry_outline_add_join(struct d2d_geometry *geometry,
     }
     f = &geometry->outline.faces[geometry->outline.face_count];
 
-    ccw = d2d_point_ccw(&origin, prev, next);
-    if (ccw == 0.0f)
-    {
-        d2d_outline_vertex_set(&v[0], p0->x, p0->y, -prev->x, -prev->y, -prev->x, -prev->y);
-        d2d_outline_vertex_set(&v[1], p0->x, p0->y,  prev->x,  prev->y,  prev->x,  prev->y);
-        d2d_outline_vertex_set(&v[2], p0->x + 25.0f * -prev->x, p0->y + 25.0f * -prev->y,
-                 prev->x,  prev->y,  prev->x,  prev->y);
-        d2d_outline_vertex_set(&v[3], p0->x + 25.0f * -prev->x, p0->y + 25.0f * -prev->y,
-                -prev->x, -prev->y, -prev->x, -prev->y);
-    }
-    else if (ccw < 0.0f)
+    if (ccw < 0.0f)
     {
         d2d_outline_vertex_set(&v[0], p0->x, p0->y,  next->x,  next->y, -prev->x, -prev->y);
         d2d_outline_vertex_set(&v[1], p0->x, p0->y, -next->x, -next->y, -next->x, -next->y);
