msvcrt: make C++ catch frame unregistration idempotent.

diff --git a/dlls/msvcrt/except.c b/dlls/msvcrt/except.c
--- a/dlls/msvcrt/except.c
+++ b/dlls/msvcrt/except.c
@@ -913,6 +913,12 @@ void CDECL _FindAndUnlinkFrame(frame_info *fi)
 
     TRACE("(%p)\n", fi);
 
+    if (!cur)
+    {
+        ERR("frame %p not found, frame list is empty\n", fi);
+        return;
+    }
+
     if (cur == fi)
     {
         data->frame_info_head = cur->next;
@@ -1007,11 +1013,13 @@ void CDECL __CxxUnregisterExceptionObject(cxx_frame_info *frame_info, BOOL in_us
         return;
 
     _FindAndUnlinkFrame(&frame_info->frame_info);
-    if(data->exc_record->ExceptionCode == CXX_EXCEPTION && !in_use
+    if(data->exc_record && data->exc_record->ExceptionCode == CXX_EXCEPTION && !in_use
             && _IsExceptionObjectToBeDestroyed((void*)data->exc_record->ExceptionInformation[1]))
         __DestructExceptionObject(data->exc_record);
     data->exc_record = frame_info->rec;
     data->ctx_record = frame_info->context;
+    frame_info->rec = (void*)-1;
+    frame_info->context = (void*)-1;
 }
 
 struct __std_exception_data {
